bcda/service/resources.go

```diff
@@ -0,0 +1,68 @@
+package service
+
+import "github.com/CMSgov/bcda-app/bcda/constants"
+
+//DataType is used to identify the type of data returned by each resource
+type DataType struct {
+	Adjudicated    bool
+	PreAdjudicated bool
+}
+
+var dataTypeMap map[string]DataType
+
+// SupportsDataType checks if the dataType is supported by the instanced DataType object
+func (r DataType) SupportsDataType(dataType string) bool {
+	switch dataType {
+	case constants.PreAdjudicated:
+		return r.PreAdjudicated
+	case constants.Adjudicated:
+		return r.Adjudicated
+	default:
+		return false
+	}
+}
+
+// init creates a map of ResourceType => DataType configurations to be used by later functions
+func init() {
+	dataTypeMap = map[string]DataType{
+		"Patient":              {Adjudicated: true},
+		"Coverage":             {Adjudicated: true},
+		"ExplanationOfBenefit": {Adjudicated: true},
+		"Observation":          {Adjudicated: true},
+		"Claim":                {Adjudicated: false, PreAdjudicated: true},
+		"ClaimResponse":        {Adjudicated: false, PreAdjudicated: true},
```

Comment:
Assuming values default to false could we just remove the Adjudicated value for Claim and ClaimResponse?